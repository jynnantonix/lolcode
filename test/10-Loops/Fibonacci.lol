HAI,		BTW Fibonacci Numbers
I HAS A MAX
I HAS A NUM ITZ 1
I HAS A NEXT ITZ 1
I HAS A COUNT ITZ 0

VISIBLE "How many fibonacci numbers do you wish to see?" MKAY?
GIMMEH MAX

MAX R MAEK MAX A NUMBR

DIFFRINT COUNT AN BIGGR OF COUNT AN MAX,	BTW if COUNT < MAX
O RLY?
    YA RLY 
        VISIBLE NEXT MKAY?
        COUNT R SUM OF COUNT AN 1,		BTW COUNT++
OIC

DIFFRINT COUNT AN BIGGR OF COUNT AN MAX,	BTW if COUNT < MAX
O RLY?
    YA RLY
        VISIBLE NEXT MKAY?
        COUNT R SUM OF COUNT AN 1,		BTW COUNT++
        NEXT R SUM OF NEXT AN NUM,		BTW NEXT = NEXT + NUM
OIC

BTW for(; COUNT < MAX; COUNT++)
IM IN YR loop WILE DIFFRINT COUNT AN BIGGR OF COUNT AN MAX
    VISIBLE NEXT MKAY?
    SUM OF NEXT AN NUM,					BTW IT = NEXT + NUM
    NUM R NEXT,							BTW NUM = NEXT
    NEXT R IT,							BTW NEXT = IT
    COUNT R SUM OF COUNT AN 1,			BTW COUNT += 1
IM OUTTA YR loop

KTHXBAI
