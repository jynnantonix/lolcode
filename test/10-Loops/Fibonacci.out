How many fibonacci numbers do you wish to see? 
LOL>> 15
1 
1 
2 
3 
5 
8 
13 
21 
34 
55 
89 
144 
233 
377 
610 
